packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js

```diff
@@ -2,17 +2,100 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useCallback } from '@wordpress/element';
+import { useCallback, useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore, useEntityBlockEditor } from '@wordpress/core-data';
+import { BlockEditorProvider, Inserter } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import SidebarNavigationScreen from '../sidebar-navigation-screen';
-import NavigationInspector from '../navigation-inspector';
 import { useHistory } from '../routes';
+import NavigationMenuContent from './navigation-menu-content';
+import SidebarButton from '../sidebar-button';
+
+const NAVIGATION_MENUS_QUERY = { per_page: -1, status: 'publish' };
+
+function SidebarNavigationScreenWrapper( { children, actions } ) {
+	return (
+		<SidebarNavigationScreen
+			title={ __( 'Navigation' ) }
+			actions={ actions }
+			content={
+				<>
+					<p className="edit-site-sidebar-navigation-screen-navigation-menus__description">
+						{ __(
+							'Browse your site, edit pages, and manage your primary navigation menu.'
+						) }
+					</p>
+					{ children }
+				</>
+			}
+		/>
+	);
+}
 
 export default function SidebarNavigationScreenNavigationMenus() {
 	const history = useHistory();
+	const { navigationMenus, hasResolvedNavigationMenus } = useSelect(
+		( select ) => {
+			const { getEntityRecords, hasFinishedResolution } =
+				select( coreStore );
+
+			const navigationMenusQuery = [
+				'postType',
+				'wp_navigation',
+				NAVIGATION_MENUS_QUERY,
+			];
+			return {
+				navigationMenus: getEntityRecords( ...navigationMenusQuery ),
+				hasResolvedNavigationMenus: hasFinishedResolution(
+					'getEntityRecords',
+					navigationMenusQuery
+				),
+			};
+		},
+		[]
+	);
+
+	// This is copied from the edit component of the Navigation block.
+	const orderedNavigationMenus = useMemo(
```

Comment:
I was wondering if we could avoid this entirely, by passing orderBy something to the query? Don't we have that already? cc @scruffian

Reply:
The status publish filter is already on the query so at least that part can already be removed.